backend/ec2.go

```diff
@@ -679,9 +685,22 @@ func (i *ec2Instance) runScriptSSH(ctx gocontext.Context, output io.Writer) (*Ru
 	return &RunResult{Completed: err != nil, ExitCode: exitStatus}, errors.Wrap(err, "error running script")
 }
 
+func (p *ec2Provider) deleteKeyPair(keyName string) error {
+	svc := ec2.New(p.awsSession)
+	deleteKeyPairInput := &ec2.DeleteKeyPairInput{
+		KeyName: aws.String(keyName),
+	}
+
+	_, err := svc.DeleteKeyPair(deleteKeyPairInput)
+
+	if err != nil {
+		return err
```

Comment:
Log before returning?

Reply:
Error is being propagated, logging is done elsewhere